core/src/main/java/nl/nn/adapterframework/stream/Message.java

```diff
@@ -354,6 +392,61 @@ public InputStream asInputStream(String defaultCharset) throws IOException {
 		}
 	}
 
+	/**
+	 * Reads the first 10k of a binary message. If the message does not support markSupported it is wrapped in a buffer.
+	 * Only works for binary messages
+	 */
+	public byte[] getMagic() throws IOException {
+		return getMagic(10*1024);
+	}
+
+	/**
+	 * Reads the first 10k of a binary message. If the message does not support markSupported it is wrapped in a buffer.
+	 * Only works for binary messages
+	 * @param readLimit amount of bytes to read.
+	 */
+	public byte[] getMagic(int readLimit) throws IOException {
```

Comment:
Wordt deze getMagic nog ergens anders gebruikt dan in MessageUtils? Moet hij niet daar heen dan?

Reply:
Nee, alleen in de MessageUtils. Hij kan niet verplaatst worden omdat de InputStream gewrapped moet worden wanneer dit geen markSupported stream is.